### PR TITLE
New backup service API version 2021-01-01

### PR DESCRIPTION
**Pull request by anjorsh**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet. Feb 2021
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet. Jan 2021
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [x] SDK of Java
      - [x] SDK of Js
      - [x] SDK of Go
      - [x] PowerShell
      - [x] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-01-01/bms.json [hunk 1]
```diff
@@ -0,0 +1,10157 @@
...
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{vaultName}/backupconfig/vaultconfig": {
```
Comment (zcxcasd) [inline specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-01-01/bms.json:21]: @pilor  backupconfig doesn't seem to be an official type in the manifest, is this ok?

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/bms.json [hunk 2]
```diff
@@ -275,10 +275,6 @@
             }
           }
         },
-        "x-ms-long-running-operation": true,
```
Comment (pilor) [inline specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/bms.json:278]: You can't make changes to existing api-versions.  If you feel these are necessary please get signoff by going through the breaking change review process in the PR description

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-01-01/bms.json (hunk 1 above)
Comment (pilor) [reply specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-01-01/bms.json:21]: child proxy types don't necessarily have to be in the manifest if they use the same endpoints as the parent

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-01-01/bms.json [hunk 3]
```diff
@@ -0,0 +1,10157 @@
...
+          }
+        },
+        "x-ms-examples": {
+          "Update Vault Security Config": {
+            "$ref": "./examples/Common/BackupResourceVaultConfigs_Put.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{vaultName}/backupEncryptionConfigs/backupResourceEncryptionConfig": {
```
Comment (zcxcasd) [inline specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-01-01/bms.json:170]: same for this, not in manifest ._.

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-01-01/bms.json [hunk 4]
```diff
@@ -0,0 +1,10157 @@
...
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/VaultStorageConfigOperationResultResponse"
+            }
+          },
+          "202": {
```
Comment (pilor) [inline specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-01-01/bms.json:638]: If this returns 202 it must be marked as x-ms-long-running-operation (it was previously)

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-01-01/bms.json [hunk 5]
```diff
@@ -0,0 +1,10157 @@
...
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/AzureVmWorkloadProtectableItem"
+        }
+      ],
+      "x-ms-discriminator-value": "SAPAseSystem"
+    },
+    "AzureVmWorkloadSAPAseSystemWorkloadItem": {
+      "descriptio
```
Comment (pilor) [inline specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-01-01/bms.json:10063]: MoveRPAcrossTiersRequestResource is the body of a POST request.  It does not make sense to have top level resource properties like "id", "name", "location", etc...

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-01-01/bms.json (hunk 3 above)
Comment (anjorsh) [reply specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-01-01/bms.json:170]: Addressed by @pilor above.

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/bms.json (hunk 2 above)
Comment (anjorsh) [reply specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/bms.json]: This breaking change has been approved [as part of another PR](https://github.com/Azure/azure-rest-api-specs/pull/12547).

I have pulled latest commits from master to remove it from this pull request.

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-01-01/bms.json (hunk 5 above)
Comment (anjorsh) [reply specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-01-01/bms.json]: Changed the request body to MoveRPAcrossTiersRequest and removed MoveRPAcrossTiersRequestResource

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-01-01/bms.json (hunk 4 above)
Comment (anjorsh) [reply specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-01-01/bms.json]: This is not meant to be a long-running-operation. Removed the 202 response as well.

Comment (anjorsh) [reply specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-01-01/bms.json:638]: This is the OperationResults API for the long-running operation above (BMSPrepareDataMove). This returns 202 if the parent BMSPrepareDataMove operation is still in progress, but is not a long running operation in itself.

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/bms.json [hunk 6]
```diff
@@ -1828,14 +1828,29 @@
...
           "type": "string",
           "x-ms-enum": {
             "name": "RestorePointType",
             "modelAsString": true
           },
           "readOnly": true
+        },
+        "recoveryPointTierDetails": {
+          "description": "Recovery point tier information.",
+          "type": "array",
```
Comment (pilor) [inline specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/bms.json:1843]: The breaking change tool may not be picking up changes in polymorphic types correctly... these are all breaking changes to an existing API-version and should go through breaking change review.

@njuCZ can you check if OAD is doing the right thing with polymorphic types?  Adding new properties to an existing version should always cause a break, yes?

Comment (cvarsh) [reply specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/bms.json:1843]: These are the optional fields added to existing version that should not cause break to the client.

Comment (pilor) [reply specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/bms.json:1843]: The official stance of the breaking change folks is no changes to existing api-versions.  Please get their approval.  Process is in the PR description

Comment (anjorsh) [reply specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/bms.json:1843]: In the PR description, the breaking change checklist is as follows, and none of these changes are included in this PR.

>  Removing API(s) in stable version
>  Removing properties in stable version
>  Removing API version(s) in stable version
>  Updating API in stable version with Breaking Change Validation errors
>  Updating API(s) in preview over 1 year

Furthermore, the description mentions below, which is what we are doing in this PR.

> “adding a new property” no more require ARM API review.

Comment (anjorsh) [reply specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/bms.json:1843]: Hi @pilor, given that the guidance in the PR description (quoted above) seems to indicate that this change should not need a breaking change review, is it still required to go through the process?

Comment (pilor) [reply specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/bms.json:1843]: Yes, these are breaking changes

Comment (anjorsh) [reply specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/bms.json]: I have removed all changes to older API versions from this PR.

Comment (anjorsh) [reply specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2018-12-20/bms.json]: I have removed all changes to previous versions from this PR. This PR now contains only the new version addition.

**Comment by anjorsh** [comment]
Thank you @pilor for review!

@ArcturusZhang/@lirenhe/@ChenTanyi: Can you please help getting this PR merged?

**Review by njuCZ — APPROVED** [review]

**Pull request merged**